packages/toast/src/toast.ts

```diff
@@ -0,0 +1,207 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    html,
+    CSSResultArray,
+    TemplateResult,
+    LitElement,
+    property,
+    PropertyValues,
+} from 'lit-element';
+import '@spectrum-web-components/button';
+import '@spectrum-web-components/icon';
+import '@spectrum-web-components/icons';
+
+import toastStyles from './toast.css.js';
+
+export const toastVariants = [
+    'negative',
+    'positive',
+    'info',
+    'error',
+    'warning',
+];
+
+/**
+ * @slot - The toast content
+ */
+
+export class Toast extends LitElement {
+    public static get styles(): CSSResultArray {
+        return [toastStyles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @property({ type: Boolean, reflect: true })
+    public disappearing = false;
+
+    /**
+     * The variant applies specific styling when set to `negative`, `positive`, `info`, `error`, or `warning`.
+     * `variant` attribute is removed when not matching one of the above.
+     *
+     * @param {String} variant
+     */
+    @property({ type: String })
+    public set variant(variant: string) {
```

Comment:
could we export a ToastVariant type for typescript users here?

Reply:
Would someone use that differently that simply changing this to?
```suggestion
    public set variant(variant: '' | 'negative' | 'positive' | 'info' | 'error' | 'warning') {
```
With better understanding of this, I can addresss other instances of this pattern in a separate PR.

Reply:
i'm open to either approach, was thinking that an exported toast variant type could be useful for clients to define for model interfaces that drive the toast view, although i think you could pull out the typing either way with some typescript trickery.